frontend/src/store/reunion.js

```diff
@@ -0,0 +1,79 @@
+import produce from "immer";
+import { createEvent } from "./evento";
+import { temaReducer } from "./tema";
+
+export const reunionEventoTypes = {
+  EMPEZAR_REUNION: "Una reunion es comenzada",
+};
+
+export const reunionEventos = {
+  comenzarReunion: (reunion) =>
+    createEvent(reunionEventoTypes.EMPEZAR_REUNION, {
+      reunion,
+      comesFromWS:true
+    }),
+};
+
+export const INITIAL_REUNION_STATE = {};
+
+export const reunionReducer = (state = INITIAL_REUNION_STATE, action) =>
+  produce(state, (draft) => {
+    draft.ultimoEventoId = action.id;
+    switch (action.type) {
+      case reunionEventoTypes.EMPEZAR_REUNION: {
+        draft.reunion = action.reunion;
+        draft.reunion.temas = action.reunion.temas
+          .map((tema) => temaReducer(tema, action))
+          .sort(compareTema);
+        break;
+      }
+
+      default:
+        if (draft.reunion?.temas) {
+          const temaIndex = draft.reunion.temas.findIndex(
+            (tema) => tema.id === action.idTema
+          );
+
+          if (temaIndex !== -1){
+            draft.reunion.temas[temaIndex] = temaReducer(
+              draft.reunion.temas[temaIndex],
+              action
+            );
+            break;
+          }
+          
+          console.error(
+            "Se recibio una accion con un idTema desconocido",
+            action
+          );
+        }
```

Comment:
Acá el flujo de este pedazo de código es medio poco intuitivo, en general lo que quiero yo es que en cuanto encuentro una situacion de error (precondicion falla) quiero falla inmediatamente, y no tener varios ifs anidados, creo que este codigo se entiende mas que cosas son precondiciones y cual es el caso feliz

(DISCLAIMER: este codigo no lo probé en un navegador asi que podria tener errores de sintaxis pero por aca va la idea)

```suggestion
        if (!draft.reunion?.temas) {
          break;
        }
        
        const temaIndex = draft.reunion.temas.findIndex(
          (tema) => tema.id === action.idTema
        );

        if (temaIndex === -1){
          console.error(
            "Se recibio una accion con un idTema desconocido",
            action
          );
          break;
        }

        draft.reunion.temas[temaIndex] = temaReducer(
            draft.reunion.temas[temaIndex],
            action
          );
          break;
```

Reply:
Exelente